lang/luasocket/Makefile

```diff
@@ -8,33 +8,61 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=luasocket
-PKG_SOURCE_VERSION:=6d5e40c324c84d9c1453ae88e0ad5bdd0a631448
-PKG_VERSION:=3.0-rc1-20130909
-PKG_RELEASE:=5
+PKG_SOURCE_DATE:=2019-04-21
+PKG_SOURCE_VERSION:=733af884f1aa18ff469bf3c4d18810e815853211
+PKG_RELEASE:=6
 
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
-PKG_MIRROR_HASH:=d2fa075d8bd026c41e0eb1a634ac2ad8115dee8abb070720e8e91fab51f86ee4
+PKG_MIRROR_HASH:=a74a291cff06ab764f7ded7e5f896614714f978d96c40177f83097306a29fa19
 PKG_SOURCE_URL:=https://github.com/diegonehab/luasocket.git
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_SOURCE_VERSION)
```

Comment:
Remove

Reply:
I tested this and the build fails.  The build fails at the patch stage:
`No file to patch.  Skipping patch.`

Reply:
I believe the actual error to be in the PKG_BUILD_DIR variable, not here.

Reply:
@neheb - If I don't define the PKG_SOURCE_SUBDIR, one gets generated for me.  The one that is defined is "luasocket-733af884f1aa18ff469bf3c4d18810e815853211/".  This is where the patch is attempted to be applied.  But the actual source ends up in "luasocket-2019-04-21-733af884/".  I'm can get the PKG_NAME and PKG_SOURCE_DATE but I don't know how to get the short sha without doing some ugly string manipulation.  I looked here https://openwrt.org/docs/guide-developer/packages and I don't see an appropriate variable to use to get $(PKG_NAME)-$(PKG_SOURCE_DATE)-SHORT_SHA to make the PKG_BUILD_DIR variable correct for the patching.  Do you know what variable I can use so PKG_BUILD_DIR is correct without defining PKG_SOURCE_SUBDIR?

Reply:
The defaults are here: https://github.com/openwrt/openwrt/blob/master/include/download.mk#L14

They work fine elsewhere. ping @yousong

Reply:
@neheb - Thanks for the link.  I was able to remove PKG_SOURCE_SUBDIR now seeing how things are done in that download.mk.